src/transformers/models/audio_spectrogram_transformer/feature_extraction_audio_spectrogram_transformer.py

```diff
@@ -230,7 +262,45 @@ def __call__(
         if self.do_normalize:
             padded_inputs["input_values"] = [self.normalize(feature) for feature in input_values]
 
+        if add_noise:
+            padded_inputs["input_values"] = [self.add_noise(feature) for feature in padded_inputs["input_values"]]
+
+        if add_temporal_shift:
+            padded_inputs["input_values"] = [
+                self.add_temporal_shift(feature) for feature in padded_inputs["input_values"]
+            ]
+
         if return_tensors is not None:
             padded_inputs = padded_inputs.convert_to_tensors(return_tensors)
 
         return padded_inputs
+
+    # from https://github.com/YuanGongND/ast with minor changes
+    def apply_time_frequency_masks(self, fbank, frequency_mask_length, time_mask_length):
+        if not is_speech_available():
+            raise ImportError("Torchaudio is required to use frequency and/or time masking.")
+        if not isinstance(fbank, torch.Tensor):
+            fbank = torch.tensor(fbank)
+        fbank = torch.transpose(fbank, 0, 1)
+        # this is just to satisfy new torchaudio version, which only accept [1, freq, time]
+        fbank = fbank.unsqueeze(0)
+        if frequency_mask_length:
+            freqm = FrequencyMasking(frequency_mask_length)
+            fbank = freqm(fbank)
+        if time_mask_length:
+            timem = TimeMasking(time_mask_length)
+            fbank = timem(fbank)
+        # squeeze it back, it is just a trick to satisfy new torchaudio version
+        fbank = fbank.squeeze(0)
+        fbank = torch.transpose(fbank, 0, 1)
+        fbank = fbank.numpy()
+        return fbank
+
+    # from https://github.com/YuanGongND/ast but converted into numpy
+    def add_noise(self, fbank: np.ndarray) -> np.ndarray:
+        fbank += np.random.rand(fbank.shape[0], fbank.shape[1]) * (np.random.rand() / 10)
+        return fbank
```

Comment:
This is a really tiny nit but you could improve the readability by splitting into multiple lines:
```suggestion
        noise = np.random.rand(fbank.shape[0], fbank.shape[1]) 
        noise = noise * (np.random.rand() / 10)
        fbank += noise
        return fbank
```